### DISCUSSION TITLE
fluentd plugin or approach to enrich event data using external REST service.

### DISCUSSION THREAD

**Discussion by mayfly-dev** [fluent/fluentd#3657] [Q&A]
Hello Team,

I am new here. 

We were using logstash for ETL now and evaluating fluentd due to its performance and still being lightweight.

Need one help, we have an enrichment step that invokes an HTTP endpoint to fetch metadata about the transaction logs.

Is there any official plugin or a way to achieve the same with fluentd ?.

Thanks

**Comment by fujimotos** [comment]
> we have an enrichment step that invokes an HTTP endpoint to fetch metadata about the transaction logs.

AFAIK, there are a few plugins that do that. For example:

* https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter
* https://github.com/takus/fluent-plugin-ec2-metadata

The caveat is that those plugins are all domein-specfic i.e. only support
particular types of endpoints.

So I think you probably need to implement one by yourself, unless the
endpoint is very popular one.
